### PR TITLE
Close bpf module on Stop

### PR DESCRIPTION
**Pull request by 2opremio**
Followup from https://github.com/iovisor/gobpf/pull/51

### CODE REVIEW

**Comment by 2opremio** [comment]
@alban I am a bit clueless about the rkt error. Is the status not being recovered properly or the tests are actually failing? (the CI output seems to suggest the former)

**Comment by alban** [comment]
SemaphoreCI says:

> only one map with size 280 allowed per section

It's due to changes in gobpf implementing the bpf map pinning feature, unrelated to the Close feature: https://github.com/iovisor/gobpf/pull/49

struct bpf_map_def needs to be updated in https://github.com/weaveworks/tcptracer-bpf/blob/master/bpf_helpers.h#L63 to match https://github.com/iovisor/gobpf/blob/master/elf/elf.go#L61

Tcptracer-bpf doesn't need bpf map pinning, so the maps definitions in https://github.com/weaveworks/tcptracer-bpf/blob/master/tcptracer-bpf.c can be left with pinning=0 and namespace="".

/cc @schu

**Comment by 2opremio** [comment]
Thanks @alban 

It seems everything went well. Mind taking a look?

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,9 @@ ENV GOPATH /go
 RUN dnf update -y vim-minimal && \
 	dnf install -y llvm clang kernel-devel make binutils vim-common golang go-bindata ShellCheck git file
 
-RUN go get -u github.com/mvdan/sh/cmd/shfmt
+RUN curl -fsSL -o shfmt https://github.com/mvdan/sh/releases/download/v1.3.0/shfmt_v1.3.0_linux_amd64 && \
+    chmod +x shfmt && \
+    mv shfmt /usr/bin
```
Comment (alban) [inline Dockerfile]: No checksum check? In equivalent code in `tools/`, there is a check.

File: tools/build/golang/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,11 @@ RUN apt-get update && \
       python-yaml \
       unzip && \
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-RUN pip install attrs
+RUN pip install attrs pyhcl
+RUN curl -fsSLo shfmt https://github.com/mvdan/sh/releases/download/v1.3.0/shfmt_v1.3.0_linux_amd64 && \
+	echo "b1925c2c405458811f0c227266402cf1868b4de529f114722c2e3a5af4ac7bb2  shfmt" | sha256sum -c && \
+	chmod +x shfmt && \
+	mv shfmt /usr/bin
```
Comment (alban) [inline tools/build/golang/Dockerfile:20]: Should sha512 be used instead?

**Comment by alban** [comment]
@2opremio  2 remarks about checksums but LGTM otherwise.

File: tools/build/golang/Dockerfile (hunk 2 above)
Comment (2opremio) [reply tools/build/golang/Dockerfile:20]: Maybe, let's create a separate PR for that if needed.

**Pull request merged**